ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -845,6 +913,9 @@ func validateRayJobSpec(rayJob *rayv1.RayJob) error {
 	if rayJob.Spec.BackoffLimit != nil && *rayJob.Spec.BackoffLimit < 0 {
 		return fmt.Errorf("backoffLimit must be a positive integer")
 	}
+	if rayJob.Spec.ShutdownAfterJobFinishes && rayJob.Spec.DeletionPolicy != nil && *rayJob.Spec.DeletionPolicy == rayv1.DeleteNoneDeletionPolicy {
```

Comment:
Add another validation for Kueue. That is, if `rayJob.Spec.Suspend` is set to true, the `DeletionPolicy` should be nil or `DeleteClusterDeletionPolicy`